pkg/controller/node/nodecontroller.go

```diff
@@ -244,6 +244,26 @@ func NewNodeController(
 					glog.Errorf("Error allocating CIDR: %v", err)
 				}
 			},
+			UpdateFunc: func(_, obj interface{}) {
+				node := obj.(*api.Node)
+				// If the PodCIDR is not empty we either:
+				// - already processed a Node that already had a CIDR after NC restarted (cidr is marked as used),
+				// - already processed a Node successfully and allocated a CIDR for it (cidr is marked as used),
+				// - already processed a Node but we did saw a "timeout" response and request eventually got through
+				//   in this case we haven't released the allocated CIDR (cidr is still marked as used).
+				// There's a possible error here:
+				// - NC sees a new Node and assigns a CIDR X to it,
+				// - Update Node call fails with a timeout,
+				// - Node is updated by some other component, NC sees an update and assigns CIDR Y to the Node,
+				// - Both CIDR X and CIDR Y are marked as used in the local cache, even though Node sees only CIDR Y
+				// Restart of NC fixes the issue.
```

Comment:
Please state explicitly that the problem is that in memory an unoccupied CIDR is marked as occupied, which prevents it from being allocated to something. But the state of the cluster itself is correct.